cpp/dolfinx/fem/Function.h

```diff
@@ -400,15 +379,43 @@ class Function
     std::int32_t num_cells = cell_imap->size_local() + cell_imap->num_ghosts();
     std::vector<std::int32_t> cells(num_cells, 0);
     std::iota(cells.begin(), cells.end(), 0);
+    interpolate(e, cells, expr_mesh, cell_map);
+  }
 
-    if (cell_map.size() == 0)
-      interpolate(e, cells, expr_mesh, cell_map);
-    else
-      interpolate(e, cells, expr_mesh, std::span(cells.data(), cells.size()));
+  /// Interpolate a function defined on a non-matching mesh
+  /// @param[in] v The function to be interpolated
+  /// @param[in] padding Absolute padding for every bounding box on the mesh of
+  /// \p this to determine point ownership
+  /// @brief Evaluate the Function at points.
+  /// @note For repeated interpolation, it is adviced to precompute the
```

Comment:
But this function doesn't accept any interpolation data. It's not clear what the caller is being advised to do.